compiler/rustc_target/src/spec/armv7_sony_vita_newlibeabihf.rs

```diff
@@ -29,7 +29,7 @@ pub fn target() -> Target {
             families: cvs!["unix"],
             linker: Some("arm-vita-eabi-gcc".into()),
             relocation_model: RelocModel::Static,
-            features: "+v7,+neon".into(),
+            features: "+v7,+neon,+vfp3,+thumb2".into(),
```

Comment:
You need to enable the `thumb_mode` feature if you want LLVM to actually generate thumb code.

Reply:
Hm, I thought `llvm_target: "thumbv7a-vita-eabihf".into()` forces to generate thumb code. I'll double check

Reply:
So, I've checked, and changing the target to `thumbv7a` does the trick, at the same time adding `+thumb_mode` to features with `armv7a` target still produces code in arm ISA with this spec.

Also, I've added `has_thumb_interworking` flag to spec, so it would be possible to explicitly set ISA from code with `#[cfg_attr(target_cpu="arm", instruction_set(arm::a32))]`